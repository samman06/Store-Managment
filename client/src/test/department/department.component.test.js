import React from "react";
import {render, shallow} from "enzyme";
import AddDepartmentModal from "../../components/Department/addDepartmentModal"
import EditDepartmentModal from "../../components/Department/editDepartmentModal"
import DepartmentItem from "../../components/Department/departmentItem"

describe("Test Department shallow", () => {
    it("should DepartmentItem length be 1", () => {
        expect(shallow(<DepartmentItem departments={[]}/>).length).toEqual(1);
    });
    it("should EditDepartmentModal length be 1", () => {
        expect(shallow(<EditDepartmentModal errors={{}}/>).length).toEqual(1);
    });
    it("should AddDepartmentModal length be 1", () => {
        expect(shallow(<AddDepartmentModal errors={{}}/>).length).toEqual(1);
    })
});

describe("Test Department Snapshot", () => {
    it("Take DepartmentItem Snapshot", () => {
        expect(shallow(<DepartmentItem departments={[]}/>)).toMatchSnapshot();
    });
    it("Take EditDepartmentModal Snapshot", () => {
        expect(shallow(<EditDepartmentModal errors={{}}/>)).toMatchSnapshot();
    });
    it("Take AddDepartmentModal Snapshot", () => {
        expect(shallow(<AddDepartmentModal errors={{}}/>)).toMatchSnapshot();
    });
});


describe("Test Department render", () => {
    it("expect DepartmentItem render with selected element", () => {
        expect(render(<DepartmentItem departments={[]}/>)["0"].name).toEqual("div");
        expect(render(<DepartmentItem departments={[]}/>)["0"].children[0].name).toEqual("div");
    });
    it("expect EditDepartmentModal render with selected element", () => {
        expect(render(<EditDepartmentModal errors={{}}/>)["0"].name).toEqual("div");
    });
    it("expect AddDepartmentModal render with selected element", () => {
        expect(render(<AddDepartmentModal errors={{}}/>)["0"].name).toEqual("div");
        expect(render(<AddDepartmentModal errors={{}}/>)["0"].children[0].name).toEqual("button");
    });
});

