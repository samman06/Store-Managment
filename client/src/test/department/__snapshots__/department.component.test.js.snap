// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Department Snapshot Take AddDepartmentModal Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <addDepartmentModal
    errors={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-info mt-1 mb-1"
          onClick={[Function]}
>
          Add Department +
</button>,
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          toggle={[Function]}
          unmountOnClose={true}
          zIndex={1050}
>
          <div
                    className="modal-header"
          >
                    Add Department
          </div>
          <div
                    className="modal-body"
          >
                    <div
                              className="form-group"
                    >
                              <input
                                        className="form-control"
                                        name="name"
                                        onChange={[Function]}
                                        placeholder="Department Name"
                                        type="name"
                              />
                    </div>
          </div>
          <div
                    className="modal-footer"
          >
                    <button
                              className="btn btn-primary mr-1"
                              onClick={[Function]}
                    >
                              Add Department
                    </button>
                    <button
                              className="btn btn-warning"
                              onClick={[Function]}
                    >
                              cancel
                    </button>
          </div>
</Modal>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add Department +",
          "className": "btn btn-info mt-1 mb-1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Add Department +",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <div
              className="modal-header"
>
              Add Department
</div>,
            <div
              className="modal-body"
>
              <div
                            className="form-group"
              >
                            <input
                                          className="form-control"
                                          name="name"
                                          onChange={[Function]}
                                          placeholder="Department Name"
                                          type="name"
                            />
              </div>
</div>,
            <div
              className="modal-footer"
>
              <button
                            className="btn btn-primary mr-1"
                            onClick={[Function]}
              >
                            Add Department
              </button>
              <button
                            className="btn btn-warning"
                            onClick={[Function]}
              >
                            cancel
              </button>
</div>,
          ],
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "returnFocusAfterClose": true,
          "role": "dialog",
          "scrollable": false,
          "toggle": [Function],
          "unmountOnClose": true,
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add Department",
              "className": "modal-header",
            },
            "ref": null,
            "rendered": "Add Department",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="form-group"
>
                <input
                                className="form-control"
                                name="name"
                                onChange={[Function]}
                                placeholder="Department Name"
                                type="name"
                />
</div>,
              "className": "modal-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="form-control"
                    name="name"
                    onChange={[Function]}
                    placeholder="Department Name"
                    type="name"
/>,
                  undefined,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "name",
                    "onChange": [Function],
                    "placeholder": "Department Name",
                    "type": "name",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                undefined,
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn btn-primary mr-1"
                  onClick={[Function]}
>
                  Add Department
</button>,
                <button
                  className="btn btn-warning"
                  onClick={[Function]}
>
                  cancel
</button>,
              ],
              "className": "modal-footer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Add Department",
                  "className": "btn btn-primary mr-1",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Add Department",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "cancel",
                  "className": "btn btn-warning",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "cancel",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-info mt-1 mb-1"
            onClick={[Function]}
>
            Add Department +
</button>,
          <Modal
            autoFocus={true}
            backdrop={true}
            backdropTransition={
                        Object {
                                    "mountOnEnter": true,
                                    "timeout": 150,
                                  }
            }
            centered={false}
            fade={true}
            isOpen={false}
            keyboard={true}
            modalTransition={
                        Object {
                                    "timeout": 300,
                                  }
            }
            onClosed={[Function]}
            onOpened={[Function]}
            returnFocusAfterClose={true}
            role="dialog"
            scrollable={false}
            toggle={[Function]}
            unmountOnClose={true}
            zIndex={1050}
>
            <div
                        className="modal-header"
            >
                        Add Department
            </div>
            <div
                        className="modal-body"
            >
                        <div
                                    className="form-group"
                        >
                                    <input
                                                className="form-control"
                                                name="name"
                                                onChange={[Function]}
                                                placeholder="Department Name"
                                                type="name"
                                    />
                        </div>
            </div>
            <div
                        className="modal-footer"
            >
                        <button
                                    className="btn btn-primary mr-1"
                                    onClick={[Function]}
                        >
                                    Add Department
                        </button>
                        <button
                                    className="btn btn-warning"
                                    onClick={[Function]}
                        >
                                    cancel
                        </button>
            </div>
</Modal>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add Department +",
            "className": "btn btn-info mt-1 mb-1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Add Department +",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "backdrop": true,
            "backdropTransition": Object {
              "mountOnEnter": true,
              "timeout": 150,
            },
            "centered": false,
            "children": Array [
              <div
                className="modal-header"
>
                Add Department
</div>,
              <div
                className="modal-body"
>
                <div
                                className="form-group"
                >
                                <input
                                                className="form-control"
                                                name="name"
                                                onChange={[Function]}
                                                placeholder="Department Name"
                                                type="name"
                                />
                </div>
</div>,
              <div
                className="modal-footer"
>
                <button
                                className="btn btn-primary mr-1"
                                onClick={[Function]}
                >
                                Add Department
                </button>
                <button
                                className="btn btn-warning"
                                onClick={[Function]}
                >
                                cancel
                </button>
</div>,
            ],
            "fade": true,
            "isOpen": false,
            "keyboard": true,
            "modalTransition": Object {
              "timeout": 300,
            },
            "onClosed": [Function],
            "onOpened": [Function],
            "returnFocusAfterClose": true,
            "role": "dialog",
            "scrollable": false,
            "toggle": [Function],
            "unmountOnClose": true,
            "zIndex": 1050,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add Department",
                "className": "modal-header",
              },
              "ref": null,
              "rendered": "Add Department",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="form-group"
>
                  <input
                                    className="form-control"
                                    name="name"
                                    onChange={[Function]}
                                    placeholder="Department Name"
                                    type="name"
                  />
</div>,
                "className": "modal-body",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      className="form-control"
                      name="name"
                      onChange={[Function]}
                      placeholder="Department Name"
                      type="name"
/>,
                    undefined,
                  ],
                  "className": "form-group",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control",
                      "name": "name",
                      "onChange": [Function],
                      "placeholder": "Department Name",
                      "type": "name",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  undefined,
                ],
                "type": "div",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn btn-primary mr-1"
                    onClick={[Function]}
>
                    Add Department
</button>,
                  <button
                    className="btn btn-warning"
                    onClick={[Function]}
>
                    cancel
</button>,
                ],
                "className": "modal-footer",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Add Department",
                    "className": "btn btn-primary mr-1",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Add Department",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "cancel",
                    "className": "btn btn-warning",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "cancel",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Test Department Snapshot Take DepartmentItem Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DepartmentItem
    departments={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="card-deck"
>
        
</div>,
      "className": "department-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "card-deck",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="card-deck"
>
          
</div>,
        "className": "department-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "card-deck",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Test Department Snapshot Take EditDepartmentModal Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EditModal
    errors={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
                Object {
                        "mountOnEnter": true,
                        "timeout": 150,
                      }
        }
        centered={false}
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
                Object {
                        "timeout": 300,
                      }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        toggle={[Function]}
        unmountOnClose={true}
        zIndex={1050}
>
        <div
                className="modal-header"
        >
                Edit Department
        </div>
        <div
                className="modal-body"
        >
                <div
                        className="form-group"
                >
                        <input
                                className="form-control"
                                name="name"
                                onChange={[Function]}
                                placeholder="Department Name"
                                type="name"
                                value={undefined}
                        />
                </div>
        </div>
        <div
                className="modal-footer"
        >
                <button
                        className="btn btn-primary"
                        onClick={[Function]}
                >
                        Edit
                </button>
                 
                <button
                        className="btn btn-warning"
                        onClick={[Function]}
                >
                        Cancel
                </button>
        </div>
</Modal>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "autoFocus": true,
        "backdrop": true,
        "backdropTransition": Object {
          "mountOnEnter": true,
          "timeout": 150,
        },
        "centered": false,
        "children": Array [
          <div
            className="modal-header"
>
            Edit Department
</div>,
          <div
            className="modal-body"
>
            <div
                        className="form-group"
            >
                        <input
                                    className="form-control"
                                    name="name"
                                    onChange={[Function]}
                                    placeholder="Department Name"
                                    type="name"
                                    value={undefined}
                        />
            </div>
</div>,
          <div
            className="modal-footer"
>
            <button
                        className="btn btn-primary"
                        onClick={[Function]}
            >
                        Edit
            </button>
             
            <button
                        className="btn btn-warning"
                        onClick={[Function]}
            >
                        Cancel
            </button>
</div>,
        ],
        "fade": true,
        "isOpen": false,
        "keyboard": true,
        "modalTransition": Object {
          "timeout": 300,
        },
        "onClosed": [Function],
        "onOpened": [Function],
        "returnFocusAfterClose": true,
        "role": "dialog",
        "scrollable": false,
        "toggle": [Function],
        "unmountOnClose": true,
        "zIndex": 1050,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Edit Department",
            "className": "modal-header",
          },
          "ref": null,
          "rendered": "Edit Department",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="form-group"
>
              <input
                            className="form-control"
                            name="name"
                            onChange={[Function]}
                            placeholder="Department Name"
                            type="name"
                            value={undefined}
              />
</div>,
            "className": "modal-body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="form-control"
                  name="name"
                  onChange={[Function]}
                  placeholder="Department Name"
                  type="name"
                  value={undefined}
/>,
                undefined,
              ],
              "className": "form-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-control",
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "Department Name",
                  "type": "name",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              undefined,
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn-primary"
                onClick={[Function]}
>
                Edit
</button>,
              " ",
              <button
                className="btn btn-warning"
                onClick={[Function]}
>
                Cancel
</button>,
            ],
            "className": "modal-footer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Edit",
                "className": "btn btn-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Edit",
              "type": "button",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cancel",
                "className": "btn btn-warning",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
                    Object {
                              "mountOnEnter": true,
                              "timeout": 150,
                            }
          }
          centered={false}
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
                    Object {
                              "timeout": 300,
                            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          toggle={[Function]}
          unmountOnClose={true}
          zIndex={1050}
>
          <div
                    className="modal-header"
          >
                    Edit Department
          </div>
          <div
                    className="modal-body"
          >
                    <div
                              className="form-group"
                    >
                              <input
                                        className="form-control"
                                        name="name"
                                        onChange={[Function]}
                                        placeholder="Department Name"
                                        type="name"
                                        value={undefined}
                              />
                    </div>
          </div>
          <div
                    className="modal-footer"
          >
                    <button
                              className="btn btn-primary"
                              onClick={[Function]}
                    >
                              Edit
                    </button>
                     
                    <button
                              className="btn btn-warning"
                              onClick={[Function]}
                    >
                              Cancel
                    </button>
          </div>
</Modal>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "backdrop": true,
          "backdropTransition": Object {
            "mountOnEnter": true,
            "timeout": 150,
          },
          "centered": false,
          "children": Array [
            <div
              className="modal-header"
>
              Edit Department
</div>,
            <div
              className="modal-body"
>
              <div
                            className="form-group"
              >
                            <input
                                          className="form-control"
                                          name="name"
                                          onChange={[Function]}
                                          placeholder="Department Name"
                                          type="name"
                                          value={undefined}
                            />
              </div>
</div>,
            <div
              className="modal-footer"
>
              <button
                            className="btn btn-primary"
                            onClick={[Function]}
              >
                            Edit
              </button>
               
              <button
                            className="btn btn-warning"
                            onClick={[Function]}
              >
                            Cancel
              </button>
</div>,
          ],
          "fade": true,
          "isOpen": false,
          "keyboard": true,
          "modalTransition": Object {
            "timeout": 300,
          },
          "onClosed": [Function],
          "onOpened": [Function],
          "returnFocusAfterClose": true,
          "role": "dialog",
          "scrollable": false,
          "toggle": [Function],
          "unmountOnClose": true,
          "zIndex": 1050,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Edit Department",
              "className": "modal-header",
            },
            "ref": null,
            "rendered": "Edit Department",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="form-group"
>
                <input
                                className="form-control"
                                name="name"
                                onChange={[Function]}
                                placeholder="Department Name"
                                type="name"
                                value={undefined}
                />
</div>,
              "className": "modal-body",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    className="form-control"
                    name="name"
                    onChange={[Function]}
                    placeholder="Department Name"
                    type="name"
                    value={undefined}
/>,
                  undefined,
                ],
                "className": "form-group",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "form-control",
                    "name": "name",
                    "onChange": [Function],
                    "placeholder": "Department Name",
                    "type": "name",
                    "value": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                undefined,
              ],
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
>
                  Edit
</button>,
                " ",
                <button
                  className="btn btn-warning"
                  onClick={[Function]}
>
                  Cancel
</button>,
              ],
              "className": "modal-footer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Edit",
                  "className": "btn btn-primary",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Edit",
                "type": "button",
              },
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Cancel",
                  "className": "btn btn-warning",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Cancel",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
